Extensions/RosettaTraining/Scripts/TensorFlow/model.py

```diff
@@ -0,0 +1,251 @@
+
+    return [hero1]
+
+  def _get_embedded_onboard_card_id(self, card_id):
+    if not self.kEnableCardIdEmbed:
+      return None
+
+    card_id = tf.to_int32(card_id, name='card_id_to_int32')
+    with tf.variable_scope("onboard", reuse=tf.AUTO_REUSE):
+      card_id_matrix = tf.get_variable(
+          'card_embed_matrix',
+          [data_reader.kMaxCardId, self.kCardIdDimension],
+          initializer=tf.zeros_initializer())
+    card_id_embed = tf.nn.embedding_lookup(
+      card_id_matrix, card_id, name='card_id_embed')
+    card_id_embed = tf.reshape(card_id_embed, [-1, self.kCardIdDimension])
+    return card_id_embed
+
+  def _model_minion(self, input_getter):
+    inputs = []
+
+    card_id = input_getter.get_next_slice(1)
+    card_id_embed = self._get_embedded_onboard_card_id(card_id)
+    if card_id_embed is not None:
+      inputs.append(card_id_embed)
+
+    features = input_getter.get_next_slice(data_reader.kMinionFeatures)
+    inputs.append(features)
+
+    minion1 = tf.layers.dense(
+        name='minion1',
+        inputs=tf.concat(inputs, 1),
+        units=self.kMinionConvolutionHidden1,
+        reuse=tf.AUTO_REUSE,
+        activation=tf.nn.relu)
+
+    return minion1
+
+  def _model_minions(self, input_getter):
+    features = []
+    for _ in range(data_reader.kMinions):
+      features.append(self._model_minion(input_getter))
+
+    return features
+
+  def _get_embedded_hand_card_id(self, card_id):
+    if not self.kEnableHandCardIdEmbed:
+      return None
+
+    card_id = tf.to_int32(card_id, name='card_id_to_int32')
+    with tf.variable_scope("current_hand", reuse=tf.AUTO_REUSE):
+      card_id_matrix = tf.get_variable(
+          'card_embed_matrix',
+          [data_reader.kMaxCardId, self.kCardIdDimension],
+          initializer=tf.zeros_initializer())
+    card_id_embed = tf.nn.embedding_lookup(
+      card_id_matrix, card_id, name='card_id_embed')
+    card_id_embed = tf.reshape(card_id_embed, [-1, self.kCardIdDimension])
+    return card_id_embed
+
+  def _model_current_hand_card(self, input_getter):
+    outputs = []
+
+    card_id = input_getter.get_next_slice(1)
+    card_id_embed = self._get_embedded_hand_card_id(card_id)
+    if card_id_embed is not None:
+      outputs.append(card_id_embed)
+
+    card_features = input_getter.get_next_slice(
+        data_reader.kCurrentHandCardFeatures)
+    card1 = tf.layers.dense(
+        name='hand_card_1',
+        inputs=card_features,
+        units=self.kHandConvolutionHidden1,
+        reuse=tf.AUTO_REUSE,
+        activation=tf.nn.relu)
+    outputs.append(card1)
+
+    return outputs
+
+  def _model_current_hand(self, input_getter):
+    outputs = []
+    outputs.append(
+        input_getter.get_next_slice(data_reader.kCurrentHandFeatures))
+
+    for _ in range(data_reader.kCurrentHandCards):
+      outputs.extend(self._model_current_hand_card(input_getter))
+
+    return outputs
+
+  def _model_board_features(self, input_getter):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/utilForever/RosettaStone/pullRequest?prid=4748929)